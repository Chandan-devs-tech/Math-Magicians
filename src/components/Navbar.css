.navbar {
    background: #ffa500;
    width: 100%;
    height: 75px;
    gap: 280px;
    display: flex;
    justify-content:space-around;
    align-items: center;  
    color: #000;  
}

.navbarHeading {
    font-size: 45px;
    font-family:'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    cursor: pointer;
}

.navitem {
    text-decoration: none;
    padding: 8px 35px;
    color: #000;
    cursor: pointer;
    font-size: 20px;
}

.bordered {
    border-right: 2px solid #000;
    border-left: 2px solid #000;
}