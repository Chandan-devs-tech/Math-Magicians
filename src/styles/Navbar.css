.navbar {
  background: #ffa500;
  width: 100%;
  height: 75px;
  gap: 280px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  color: #fff;
}

.navbarHeading {
  font-size: 45px;
  font-family:
    "Trebuchet MS",
    "Lucida Sans Unicode",
    "Lucida Grande",
    "Lucida Sans",
    Arial,
    sans-serif;
  cursor: pointer;
}

.navitem {
  text-decoration: none;
  padding: 8px 35px;
  color: #fff;
  cursor: pointer;
  font-size: 20px;
}

.bordered {
  border-right: 2px solid #fff;
  border-left: 2px solid #fff;
}

@media screen and (max-width: 768px) {
  .navbar {
    gap: 0;
    display: flex;
    align-items: center;
    color: #fff;
    flex-direction: column;
    width: auto;
    height: 125px;
  }
}
