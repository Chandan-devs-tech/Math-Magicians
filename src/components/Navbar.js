import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="navbarHeading">Math Magicians</h1>
    <div className="navitems">
      <NavLink to="/" className="navitem">Home</NavLink>
      <NavLink to="/calculator" className="navitem bordered">Calculator</NavLink>
      <NavLink to="/quotes" className="navitem">Quote</NavLink>
    </div>
  </nav>
);

export default Navbar;
