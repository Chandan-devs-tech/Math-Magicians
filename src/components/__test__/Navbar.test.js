import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar', () => {
  test('should render a component', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );

    // Use toMatchSnapshot() to create or compare snapshots
    expect(container).toMatchSnapshot();
  });
});
