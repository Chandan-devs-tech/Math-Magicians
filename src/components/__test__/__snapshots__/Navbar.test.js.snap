// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render a component 1`] = `
<div>
  <nav
    class="navbar"
  >
    <h1
      class="navbarHeading"
    >
      Math Magicians
    </h1>
    <div
      class="navitems"
    >
      <a
        aria-current="page"
        class="navitem active"
        href="/"
      >
        Home
      </a>
      <a
        class="navitem bordered"
        href="/calculator"
      >
        Calculator
      </a>
      <a
        class="navitem"
        href="/quotes"
      >
        Quote
      </a>
    </div>
  </nav>
</div>
`;
